src/ngraph/pass/constant_folding.cpp

```diff
@@ -231,3 +233,70 @@ void ngraph::pass::ConstantFolding::construct_constant_broadcast()
     auto broadcast_matcher = make_shared<pattern::Matcher>(broadcast, constant_broadcast_callback);
     this->add_matcher(broadcast_matcher);
 }
+
+template <class QUANT, class REAL>
+shared_ptr<op::Constant> make_constant_dequantize(shared_ptr<op::Constant> constant,
+                                                  shared_ptr<op::Dequantize> dequant,
+                                                  shared_ptr<op::Constant> scale,
+                                                  shared_ptr<op::Constant> offset)
+{
+    auto out_shape = constant->get_shape();
+    vector<REAL> out_vec(shape_size(out_shape));
+
+    runtime::reference::dequantize<QUANT, REAL>(constant->get_vector<QUANT>().data(),
+                                                scale->get_vector<REAL>().data(),
+                                                offset->get_vector<QUANT>().data(),
+                                                out_vec.data(),
+                                                constant->get_shape(),
+                                                scale->get_shape(),
+                                                dequant->get_axes());
+
+    return make_shared<op::Constant>(dequant->get_element_type(), out_shape, out_vec);
+}
+
+void ngraph::pass::ConstantFolding::construct_constant_dequantize()
+{
+    auto constant_label =
+        make_shared<pattern::op::Label>(element::u8, Shape{2}, pattern::has_class<op::Constant>());
+    auto dq_scale = op::Constant::create(element::f32, Shape{}, {1});
+    auto dq_offset = op::Constant::create(element::u8, Shape{}, {1});
+    auto dequant_op =
+        make_shared<op::Dequantize>(constant_label, dq_scale, dq_offset, element::f32, AxisSet{});
+    auto dequant = make_shared<pattern::op::Label>(dequant_op, nullptr, NodeVector{dequant_op});
```

Comment:
nitpick: this is a bit redundant ; this `m.get_match_root()` will give you yr dequantized match